src/Controls/PlanarControls.js

```diff
@@ -501,51 +510,81 @@ class PlanarControls extends THREE.EventDispatcher {
+            const zoomFactor = delta > 0 ? this.zoomInFactor : -1 * this.zoomOutFactor;
+
+            // change the camera field of view if the camera is orthographic
             if (this.camera.isOrthographicCamera) {
+                // switch state to STATE.ZOOM
+                this.state = STATE.ZOOM;
+                this.view.notifyChange(this.camera);
+
+                // camera zoom at the beginning of zoom movement
                 startZoom = this.camera.zoom;
-                endZoom = startZoom * (1 - this.zoomOutFactor);
+                // camera zoom at the end of zoom movement
+                endZoom = startZoom * (1 + zoomFactor);
+                // the altitude of the target must be the same as camera's
                 pointUnderCursor.z = this.camera.position.z;
+
+                travelAlpha = 0;
+                travelDuration = this.zoomTravelTime;
+                this.updateMouseCursorType();
+            } else {
+                // target position
+                newPos.lerpVectors(
+                    this.camera.position,
+                    pointUnderCursor,
+                    zoomFactor,
+                );
+                // initiate travel
+                this.initiateTravel(
+                    newPos,
+                    this.zoomTravelTime,
+                    null,
+                    false,
+                );
             }
-            // target position
-            newPos.lerpVectors(
-                this.camera.position,
-                pointUnderCursor,
-                -1 * this.zoomOutFactor,
-            );
-            // initiate travel
-            this.initiateTravel(
-                newPos,
-                this.zoomTravelTime,
-                null,
-                false,
-            );
+        }
+    }
+
+    /**
+     * Handle the animated zoom change for an orthographic camera, when state is `ZOOM`.
+     *
+     * @param   {number}    dt  the delta time between two updates in milliseconds
+     * @ignore
+     */
+    handleZoom(dt) {
+        travelAlpha = Math.min(travelAlpha + (dt / 1000) / travelDuration, 1);
+
+        // new zoom
+        const zoom = startZoom + travelAlpha * (endZoom - startZoom);
+
+        if (this.camera.zoom !== zoom) {
+            // zoom has changed
+            this.camera.zoom = zoom;
+            this.camera.updateProjectionMatrix();
+
+            // current world coordinates under the mouse
+            this.view.viewToNormalizedCoords(mousePosition, vect);
+            vect.z = 0;
+            this.camera.updateProjectionMatrix();
+            vect.unproject(this.camera);
+
+            // new camera position
+            this.camera.position.x += pointUnderCursor.x - vect.x;
+            this.camera.position.y += pointUnderCursor.y - vect.y;
+
+            this.camera.updateMatrixWorld(true);
+        }
+
+        // completion test
+        if (travelAlpha === 1) {
```

Comment:
this test already exists in other function?

Reply:
This test is also computed in `handleTravel`. Do you prefer it to be factorized as a method (something like `testAnimationEnd()`) ?